### PR TITLE
charasay: init at 2.0.0

### PR DESCRIPTION
**Pull request by hmajid2301**
###### Description of changes

The future of cowsay - Colorful characters saying something

Link: https://github.com/latipun7/charasay#charasay-

###### Things done

- Built on platform(s)
  -  [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

I tested the binary works locally after building it.

### CODE REVIEW

File: pkgs/tools/misc/charasay/default.nix
Comment (figsoda) [inline pkgs/tools/misc/charasay/default.nix:1]: please add this to `all-packages.nix`

**Review by figsoda — COMMENTED** [review]
please rebase your commits into 2

```
maintainers: add hmajid2301
charasay: init at 2.0.0
```

File: pkgs/tools/misc/charasay/default.nix
Comment (hmajid2301) [reply pkgs/tools/misc/charasay/default.nix:1]: okay will do!

File: pkgs/tools/misc/charasay/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  version = "2.0.0";
+
+  src = fetchFromGitHub {
+    owner = "latipun7";
+    repo = pname;
+    rev = "${version}";
+    hash = "sha256-99lMXgSHgxKc0GHnRRciMoZ+rQJyMAx+27fj6NkXxds=";
+  };
+
+  cargoHash = "sha256-bLzk/IgQniu04VZCaaCETEQxLtesMtJuBBWezYecN0A=";
```
Comment (figsoda) [inline pkgs/tools/misc/charasay/default.nix:18]:
```suggestion
    rev = "v${version}";
    hash = "sha256-99lMXgSHgxKc0GHnRRciMoZ+rQJyMAx+27fj6NkXxds=";
  };

  cargoHash = "sha256-0la16XinseOXPH2mvdYD7ZquvF2dju4UPBwl5VrTEZA=";
```

**Comment by hmajid2301** [comment]
shall i rebase again?

**Comment by figsoda** [comment]
yes, thanks

**Comment by hmajid2301** [comment]
done!

File: pkgs/tools/misc/charasay/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    owner = "latipun7";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-99lMXgSHgxKc0GHnRRciMoZ+rQJyMAx+27fj6NkXxds=";
+  };
+
+  cargoHash = "sha256-0la16XinseOXPH2mvdYD7ZquvF2dju4UPBwl5VrTEZA=";
+
+  nativeBuildInputs = [ installShellFiles ];
+
```
Comment (figsoda) [inline pkgs/tools/misc/charasay/default.nix:21]: this should fix aarch64-linux

```suggestion

  postPatch = ''
    rm .cargo/config
  '';

```

**Review by figsoda — COMMENTED** [review]
I think you rebased it into the wrong commit

btw the maintainers commit should go first

**Comment by hmajid2301** [comment]
hopefully that should be done properly :see_no_evil:. Appreciate the patience!

File: pkgs/tools/misc/charasay/default.nix (hunk 2 above)
Comment (figsoda) [reply pkgs/tools/misc/charasay/default.nix:21]: :)

File: pkgs/tools/misc/charasay/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  postPatch = ''
+    rm .cargo/config
```
Comment (figsoda) [inline pkgs/tools/misc/charasay/default.nix:23]: sorry, I had a typo

```suggestion
    rm .cargo/config.toml
```

**Review by figsoda — APPROVED** [review]

File: pkgs/tools/misc/charasay/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  cargoHash = "sha256-0la16XinseOXPH2mvdYD7ZquvF2dju4UPBwl5VrTEZA=";
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postPatch = ''
+    rm .cargo/config.toml
+  '';
+
+  postInstall = ''
+    installShellCompletion --cmd himalaya \
```
Comment (jcaesar) [inline pkgs/tools/misc/charasay/default.nix:27]: ehm. himalaya? (I' know, I'm late to the party.)

Comment (hmajid2301) [reply pkgs/tools/misc/charasay/default.nix:27]: bad copy post I think :see_no_evil:

**Pull request merged**